tests/testHtmIndex.py

```diff
@@ -176,15 +179,20 @@ def testSanity(self):
         self.assertGreater(numWithSources, 0)
 
     def testAgainstPersisted(self):
-        pix_id = 671901
+        pix_id = 2222
         dataset_name = IngestIndexedReferenceTask.ConfigClass().dataset_config.ref_dataset_name
         data_id = self.indexer.make_data_id(pix_id, dataset_name)
         self.assertTrue(self.test_butler.datasetExists('ref_cat', data_id))
         ref_cat = self.test_butler.get('ref_cat', data_id)
+        if REGENERATE_COMPARISON:
+            os.unlink(self.test_cat_path)
+            ref_cat.writeFits(self.test_cat_path)
+            return
         ex1 = ref_cat.extract('*')
         ex2 = self.test_cat.extract('*')
-        # compare sets as the order may be different
-        self.assertDictEqual(ex1, ex2)
+        self.assertEqual(set(ex1.keys()), set(ex2.keys()))
+        for kk in ex1:
+            self.assertTrue(np.all(ex1[kk] == ex2[kk]))
```

Comment:
What is the benefit to this change? The original worked, though I'd be inclined to change `assertDictEqual` to `assertEqual` (the latter should call the former, both in Python 2.7 and Python 3). Does your new code give better error messages?

If you do want to use numpy to test each key individually, then I suggest using `numpy.testing.assert_equal`, as it's shorter and more explicit.